### PR TITLE
Replace indescent words with respectful ones. Thank you Lint

### PR DESCRIPTION
**Pull request by thearifismail**
Creating this PR to replace the word "dummy" with "sample" and "whitelist" with "allowlist".

### CODE REVIEW

**Review by fpjrh — APPROVED** [review]

File: controllers/config/parameters.go [hunk 1]
```diff
@@ -501,7 +501,7 @@ func NewXJoinConfiguration() Parameters {
 				"database.server.name": "{{.ResourceNamePrefix}}.{{.Version}}",
 				"database.sslmode": "{{.HBIDBSSLMode}}",
 				"database.sslrootcert": "{{.HBIDBSSLRootCert}}",
-				"table.whitelist": "public.hosts",
+				"table.allowlist": "public.hosts",
```
Comment (fijshion) [inline controllers/config/parameters.go:504]: is table.allowlist a valid debezium option?

File: controllers/test/data/crd/strimzi-crds-0.30.0.yaml [hunk 2]
```diff
@@ -9407,7 +9407,7 @@ spec:
                     x-kubernetes-preserve-unknown-fields: true
                     type: object
                 description: CPU and memory resources to reserve.
-              whitelist:
+              allowlist:
```
Comment (fijshion) [inline controllers/test/data/crd/strimzi-crds-0.30.0.yaml:9410]: I'm not sure if these CRDs are still used but did you validate Allowlist works with strimzi?

File: pr_check.sh [hunk 3]
```diff
@@ -30,8 +30,8 @@ source $APP_ROOT/unit_test.sh
 ARTIFACTS_DIR=$WORKSPACE/artifacts
 
 mkdir -p $ARTIFACTS_DIR
-cat << EOF > $ARTIFACTS_DIR/junit-dummy.xml
+cat << EOF > $ARTIFACTS_DIR/junit-sample.xml
 <testsuite tests="1">
-    <testcase classname="dummy" name="dummytest"/>
+    <testcase classname="sample" name="sampletest"/>
```
Comment (fijshion) [inline pr_check.sh:35]: Is this allowed in our test environment?

File: controllers/test/xjoinpipeline_controller_test.go [hunk 4]
```diff
@@ -2,6 +2,10 @@ package test
 
 import (
 	"fmt"
+	"reflect"
```
Comment (CodeHeeler) [inline controllers/test/xjoinpipeline_controller_test.go:5]: Why were these three lines (reflect, strconv, time) added to the import in the same PR as the wording update?

**Pull request merged**